domain/cert/certificate.go

```diff
@@ -0,0 +1,82 @@
+package cert
+
+import (
+	"errors"
+	"time"
+
+	"github.com/satori/go.uuid"
+)
+
+// NewCertificate creates a Certificate and ensures that the activity properly set empty
+func NewCertificate(category string, activity *Activity) (*Certificate, error) {
+	if err := validate(activity); err != nil {
+		return nil, err
+	}
+	uid, err := uuid.NewV4() // TODO remove dependency to uuid lib
+	if err != nil {
+		return nil, err
+	}
+	activity.category = category
+	c := &Certificate{
+		id:       ID(uid),
+		activity: *activity,
+	}
+	return c, nil
+}
+
+// ID is a globally unique ID. Perhaps a UUID?
+type ID interface{}
+
+// A Certificate certifies an Activity and belongs to an owner (identified by any id)
```

Comment:
This comment is duplication - the property `ownerID` speaks for itself.

If the certificate is JUST for activities, instead of commenting please rather rename it to `ActivityCertificate` to be more clear.